newrelic/hooks/external_pyzeebe.py

```diff
@@ -0,0 +1,129 @@
+from newrelic.api.function_trace import FunctionTrace
+from newrelic.api.time_trace import add_custom_span_attribute
+from newrelic.api.transaction import add_custom_attribute, current_transaction
+from newrelic.common.object_wrapper import wrap_function_wrapper
+
+_logger = logging.getLogger(__name__)
+
+CLIENT_ATTRIBUTES_WARNING_LOG_MSG = "Exception occurred in PyZeebe instrumentation: Failed to add client attributes."
+
+
+# Adds client method params as txn or span attributes
+def _add_client_input_attributes(method_name, txn, args, kwargs):
+    try:
+        if method_name in ("run_process", "run_process_with_result"):
+            bpmn_id = kwargs.get("bpmn_process_id", args[0] if args else None)
+            if bpmn_id:
+                txn._add_agent_attribute("zeebe.client.bpmnProcessId", bpmn_id)
+                # add_attr("zeebe.client.bpmnProcessId", bpmn_id)
+        elif method_name == "publish_message":
+            msg_name = kwargs.get("name", args[0] if args else None)
+            if msg_name:
+                txn._add_agent_attribute("zeebe.client.messageName", msg_name)
+                # add_attr("zeebe.client.messageName", msg_name)
+            correlation_key = kwargs.get("correlation_key", args[1] if args and len(args) > 1 else None)
+            if correlation_key:
+                txn._add_agent_attribute("zeebe.client.correlationKey", correlation_key)
+                # add_attr("zeebe.client.correlationKey", correlation_key)
+            message_id = kwargs.get("message_id")
+            if message_id and len(args) > 4:
+                message_id = args[4]
+            if message_id:
+                txn._add_agent_attribute("zeebe.client.messageId", message_id)
+                # add_attr("zeebe.client.messageId", message_id)
+        elif method_name == "deploy_resource":
+            resources = list(args)
+            if len(resources) == 1 and isinstance(resources[0], (list, tuple)):
+                resources = list(resources[0])
+            if resources:
+                txn._add_agent_attribute("zeebe.client.resourceCount", len(resources))
+                # add_attr("zeebe.client.resourceCount", len(resources))
+                if len(resources) == 1:
+                    txn._add_agent_attribute("zeebe.client.resourceFile", str(resources[0]))
+                    # add_attr("zeebe.client.resourceFile", str(resources[0]))
+    except Exception:
+        _logger.warning(CLIENT_ATTRIBUTES_WARNING_LOG_MSG, exc_info=True)
+
+
+# Async wrapper that instruments router/worker annotations`
+async def _nr_wrapper_execute_one_job(wrapped, instance, args, kwargs):
+    job = args[0] if args else kwargs.get("job")
+    process_id = getattr(job, "bpmn_process_id", None) or "UnknownProcess"
+    task_type = getattr(job, "type", None) or "UnknownType"
+    txn_name = f"{process_id}/{task_type}"
+
+    with BackgroundTask(application_instance(), txn_name, group="ZeebeTask"):
```

Comment:
We may want to swap the background task here with an actual `WebTransaction`.

Reply:
Pyzeebe Workers/Tasks communicate via `gRPC` and are basically asyncio background tasks - wouldn't `BackgroundTask` be the appropriate type to use here?